x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -108,11 +137,8 @@ export class TaskRunnerFactory {
             relatedSavedObjects,
           },
           references,
-        } = await getActionTaskParams(
-          actionTaskExecutorParams,
-          encryptedSavedObjectsClient,
-          spaceIdToNamespace
-        );
+        } = taskParams!;
```

Comment:
Is it possible `beforeRun` fails and return `{ error }` so `taskParams` is undefined and using `!` will cause an error?

Reply:
getActionTaskParams failures was not handled before either, i didn't know to handle the and left as it was...

Now changed it and throw a RetryableError, hope that works.